### PR TITLE
S3 driver

### PR DESCRIPTION
**Pull request by AntonBazhal**

### CODE REVIEW

File: lib/s3.js [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+      const aws = AWS || awsSDK;
+      this._client = new aws.S3(clientOptions);
+    }
+
+    this._conf = conf;
+    this._logger = logMeGently({ level });
+  }
+
+  get client() {
```
Comment (dougmoscrop) [inline lib/s3.js]: If you actually want a private varaible, you can use Object.defineProperties instead

File: lib/s3.js [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+    const getParams = {
+      Bucket: bucket,
+      Expires: urlTTL,
+      Key: key
+    };
+
+    this._logger.log(context, { getParams, operation }, 'generating signed URL');
+    return new Promise((resolve, reject) => {
+      this._client.getSignedUrl(operation, getParams, (err, res) => {
```
Comment (dougmoscrop) [inline lib/s3.js]: just sanity check, this is because it has no .promise?

**Comment by dougmoscrop** [comment]
not as part of pr but would be nice to have like

.createUploadStream()

see: https://stackoverflow.com/questions/37336050/pipe-a-stream-to-s3-upload

File: lib/s3.js (hunk 2 above)
Comment (AntonBazhal) [reply lib/s3.js]: yes

**Pull request merged**